src/libsync/clientsideencryption.cpp

```diff
@@ -1260,6 +1335,38 @@ void ClientSideEncryption::getPublicKeyFromServer(const AccountPtr &account)
     job->start();
 }
 
+void ClientSideEncryption::getServerPublicKeyFromServer(const AccountPtr &account)
```

Comment:
@mgallien this method name is misleading. It is expected for methods starting with `get` to return something. Instead, it is returning **void** and is performing checks inside it. I suggest you either rename it to something `fetchAndCheckServerPublicKey` or `fetchAndValidatePublicKeyFromServer`, or, to make it return something, and then, using the result in further logic.

P.S. I can see that this naming is selected based on existing methods that are also named with `get` prefix while returning nothing. So, you can probably keep that name, but, also, please add `validate` or `check` into a function name, so, we could know that it is not just getting but is also validating the key. Also, we already have `getPublicKeyFromServer` which is almost exactly the same name.